activesupport/lib/active_support/notifications/instrumenter.rb

```diff
@@ -49,7 +49,7 @@ class Event
 
       def initialize(name, start, ending, transaction_id, payload)
         @name           = name
-        @payload        = payload.dup
+        @payload        = payload
```

Comment:
Can we guarantee the payload is not going to be changed down the road? Otherwise duping is unfortunately necessary.